### PR TITLE
teach GetIntegerLiteralValueString about binary literals

### PR DESCRIPTION
**Pull request by TyOverby**
closes https://github.com/dotnet/roslyn/issues/8436

### CODE REVIEW

**Comment by TyOverby** [comment]
I'm having some trouble coming up with a test case (mostly because I don't know what the ReduceTokensCodeCleanupProvider is supposed to do) , so suggestions would be very welcome!

**Comment by TyOverby** [comment]
Ping: @dotnet/roslyn-ide @dotnet/roslyn-compiler

**Comment by jasonmalinowski** [comment]
The tests are in a file called ReduceTokenTests.cs. We should have a test added before merging, but the code looks fine.

**Comment by jaredpar** [comment]
Any reason why we haven't merged this PR yet?

**Comment by TyOverby** [comment]
simple test added

**Comment by TyOverby** [comment]
@dotnet/roslyn-compiler One more review please!  (3-line diff)

File: src/Workspaces/VisualBasic/Portable/CodeCleanup/Providers/ReduceTokensCodeCleanupProvider.vb [hunk 1]
```diff
@@ -273,6 +273,9 @@ Namespace Microsoft.CodeAnalysis.CodeCleanup.Providers
                     Case LiteralBase.Octal
                         Dim val1 As ULong = ConvertToULong(value)
                         Return "&O" + ConvertToOctalString(val1)
+                    Case LiteralBase.Binary
+                        Dim asLong = CType(ConvertToULong(value), Long)
```
Comment (AlekseyTs) [inline src/Workspaces/VisualBasic/Portable/CodeCleanup/Providers/ReduceTokensCodeCleanupProvider.vb]: Do we have tests for negative SByte, Short and Integer numbers?

**Pull request merged**